spyder_kernels/comms/commbase.py

```diff
@@ -0,0 +1,407 @@
+            self._comms[comm_id].send(msg_dict, buffers=buffers)
+
+    @property
+    def _comm_name(self):
+        """
+        Get the name used for the underlying comms.
+        """
+        return 'spyder_api'
+
+    def _register_message_handler(self, message_id, handler):
+        """
+        Register a message handler.
+
+        Parameters
+        ----------
+        message_id : str
+            The identifier for the message
+        handler : callback
+            A function to handle the message. This is called with 3 arguments:
+                - msg_dict: A dictionary with message information.
+                - buffer: The data transmitted in the buffer
+                - load_exception: Exception from buffer unpickling
+            Pass None to unregister the message_id
+        """
+        if handler is None:
+            self._message_handlers.pop(message_id, None)
+            return
+
+        self._message_handlers[message_id] = handler
+
+    def _register_comm(self, comm):
+        """
+        Open a new comm to the kernel.
+        """
+        comm.on_msg(self._comm_message)
+        comm.on_close(self._comm_close)
+        self._comms[comm.comm_id] = comm
+
+    def _comm_close(self, msg):
+        """Close comm."""
+        comm_id = msg['content']['comm_id']
+        del self._comms[comm_id]
+
+    def _comm_message(self, msg):
+        """
+        Handle internal spyder messages.
+        """
+        self.calling_comm_id = msg['content']['comm_id']
+        # Load the buffer. Only one is supported.
+        try:
+            if PY2:
+                buffer = cloudpickle.loads(msg['buffers'][0])
+            else:
+                buffer = cloudpickle.loads(bytes(msg['buffers'][0]))
+            load_exception = None
+        except Exception as e:
+            load_exception = e
+            buffer = None
+
+        # Get message dict
+        msg_dict = msg['content']['data']
+
+        spyder_msg_type = msg_dict['spyder_msg_type']
+
+        if spyder_msg_type in self._message_handlers:
+            self._message_handlers[spyder_msg_type](
+                msg_dict, buffer, load_exception)
+            return
+
+        logger.debug("No such spyder message type: %s" % spyder_msg_type)
+
+    def _handle_remote_call(self, msg, buffer, load_exception):
+        """Handle a remote call."""
+        msg_dict = msg['content']
+        if load_exception:
+            logger.debug(
+                "Exception in cloudpickle.loads : %s" % str(load_exception))
+            return
+        try:
+            return_value = self._remote_callback(
+                    msg_dict['call_name'],
+                    buffer['call_args'],
+                    buffer['call_kwargs'])
+            self._set_call_return_value(msg_dict, return_value)
+        except Exception as e:
+            tb = traceback.extract_tb(sys.exc_info()[2])
+            self._set_call_return_value(msg_dict, e, traceback=tb)
```

Comment:
Are you doing something with the traceback on the Spyder side?

Reply:
This is COmmbase so it applies for both sides. The idea is to send it to the other side to be handled.

Reply:
So if the frontend calls the kernel, and an exception happens in the kernel, the exception is snet back to the frontend to be handled.

Reply:
Sure, I understand that. For now, I think the only exceptions we need to show in the frontend are those generated when calling `get_value` (as I mentioned in PR spyder-ide/spyder#9343). The others are not so important for now.

Reply:
Ok but should I then add a try: except: in the other calls? I think it makes more sense than not raising this exception.

Reply:
But that needs to be done in the frontend side, right?

Reply:
Yes, any code that calls a function that can throw an error needs to know what to do in cae of an error, or pass the error up in the stack. Here, 'up in the stack' means to the other side of the connection, so that is where this error goes.